File: android-native/app/src/main/kotlin/com/sabq/smart/feature/worldcup/WorldCupComponents.kt
package com.sabq.smart.feature.worldcup

import androidx.compose.animation.core.LinearEasing
import androidx.compose.animation.core.RepeatMode
import androidx.compose.animation.core.animateFloat
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rememberInfiniteTransition
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableLongStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage
import com.sabq.smart.ui.theme.SabqTheme
import kotlinx.coroutines.delay

/** شعار منتخب داخل دائرة بيضاء (الشعارات شفافة فتحتاج خلفية). */
@Composable
fun WcTeamLogo(team: WcTeam, size: Int = 40, ring: Color = WcColors.cardStroke) {
    Box(
        modifier = Modifier
            .size(size.dp)
            .clip(CircleShape)
            .background(Color.White)
            .border(2.dp, ring, CircleShape)
            .padding((size * 0.14).dp),
        contentAlignment = Alignment.Center,
    ) {
        AsyncImage(
            model = team.logo,
            contentDescription = team.name,
            contentScale = ContentScale.Fit,
            modifier = Modifier.fillMaxWidth(),
        )
    }
}

/** صورة دائرية للاعب (صورة/بديل). */
@Composable
fun WcPlayerPhoto(url: String, fallback: String = "", size: Int = 36) {
    if (url.isBlank()) {
        Box(
            modifier = Modifier.size(size.dp).clip(CircleShape).background(WcColors.chipFill),
            contentAlignment = Alignment.Center,
        ) {
            if (fallback.isNotBlank()) {
                Text(fallback.take(2), color = WcColors.onDarkDim, fontWeight = FontWeight.Black, fontSize = (size / 2.4).sp)
            }
        }
    } else {
        AsyncImage(
            model = url,
            contentDescription = null,
            contentScale = ContentScale.Crop,
            modifier = Modifier.size(size.dp).clip(CircleShape).background(WcColors.chipFill),
        )
    }
}

/** شارة حالة المباراة: مباشر (نبض) / انتهت / وقت الانطلاق. */
@Composable
fun WcStatusPill(fixture: WcFixture, onDark: Boolean = true) {
    when {
        fixture.status.live -> {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(4.dp),
                modifier = Modifier
                    .clip(RoundedCornerShape(50))
                    .background(WcColors.liveRed)
                    .padding(horizontal = 8.dp, vertical = 3.dp),
            ) {
                LivePulseDot()
                Text(
                    fixture.status.elapsed?.let { "$it'" } ?: fixture.status.label,
                    color = Color.White, fontSize = 11.sp, fontWeight = FontWeight.Bold,
                )
            }
        }
        fixture.status.finished -> {
            Text(
                fixture.status.label,
                color = if (onDark) Color.White.copy(alpha = 0.85f) else SabqTheme.colors.secondaryInk,
                fontSize = 11.sp, fontWeight = FontWeight.SemiBold,
                modifier = Modifier
                    .clip(RoundedCornerShape(50))
                    .background(if (onDark) Color.White.copy(alpha = 0.12f) else SabqTheme.colors.outline)
                    .padding(horizontal = 8.dp, vertical = 3.dp),
            )
        }
        else -> {
            Text(
                WcFormat.time(fixture),
                color = WcColors.emeraldDeep, fontSize = 11.sp, fontWeight = FontWeight.Bold,
                modifier = Modifier
                    .clip(RoundedCornerShape(50))
                    .background(WcColors.emerald.copy(alpha = 0.16f))
                    .padding(horizontal = 8.dp, vertical = 3.dp),
            )
        }
    }
}

@Composable
private fun LivePulseDot() {
    val transition = rememberInfiniteTransition(label = "pulse")
    val alpha by transition.animateFloat(
        initialValue = 1f, targetValue = 0.3f,
        animationSpec = infiniteRepeatable(tween(700, easing = LinearEasing), RepeatMode.Reverse),
        label = "a",
    )
    Box(modifier = Modifier.size(5.dp).clip(CircleShape).background(Color.White.copy(alpha = alpha)))
}

/** مؤقّت حيّ يعيد ملي ثانية الآن كل ثانية. */
@Composable
fun rememberSecondTicker(): Long {
    var now by remember { mutableLongStateOf(System.currentTimeMillis()) }
    LaunchedEffect(Unit) {
        while (true) {
            delay(1000)
            now = System.currentTimeMillis()
        }
    }
    return now
}

/** شرائح العدّ التنازلي (يوم/ساعة/دقيقة/ثانية) المتحرّكة. */
@Composable
fun WcCountdownChips(timestamp: Int) {
    val now = rememberSecondTicker()
    val total = (timestamp.toLong() * 1000L - now).coerceAtLeast(0L) / 1000L
    if (total <= 0L) {
        // الموعد حان والمزود لم يرفع إشارة «حية» بعد — لا أصفار مجمدة
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            Box(
                modifier = Modifier
                    .size(8.dp)
                    .clip(CircleShape)
                    .background(WcColors.emerald),
            )
            Text(
                "حان موعد الانطلاق — التغطية الحية تبدأ خلال لحظات",
                color = Color.White,
                fontSize = 13.sp,
                fontWeight = FontWeight.Bold,
            )
        }
        return
    }
    val days = (total / 86_400).toInt()
    val hours = ((total % 86_400) / 3_600).toInt()
    val minutes = ((total % 3_600) / 60).toInt()
    val seconds = (total % 60).toInt()
    Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
        chip(days, "يوم"); chip(hours, "ساعة"); chip(minutes, "دقيقة"); chip(seconds, "ثانية")
    }
}

@Composable
private fun chip(value: Int, label: String) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        modifier = Modifier
            .width(54.dp)
            .clip(RoundedCornerShape(12.dp))
            .background(Color.White.copy(alpha = 0.10f))
            .padding(vertical = 6.dp),
    ) {
        Text("$value", color = Color.White, fontSize = 20.sp, fontWeight = FontWeight.Black)
        Text(label, color = WcColors.emerald.copy(alpha = 0.85f), fontSize = 10.sp)
    }
}

/** شريط احتمالات الفوز الثلاثي. */
@Composable
fun WcProbabilityBar(fixture: WcFixture, prediction: WcPrediction) {
    val total = (prediction.home + prediction.draw + prediction.away).coerceAtLeast(1)
    val h = prediction.home * 100 / total
    val d = prediction.draw * 100 / total
    val a = prediction.away * 100 / total
    Column(modifier = Modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(6.dp)) {
        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
            Text("فوز ${fixture.home.name} $h%", color = WcColors.emerald.copy(alpha = 0.9f), fontSize = 11.sp, fontWeight = FontWeight.SemiBold)
            Text("تعادل $d%", color = Color.White.copy(alpha = 0.6f), fontSize = 11.sp, fontWeight = FontWeight.SemiBold)
            Text("فوز ${fixture.away.name} $a%", color = WcColors.emerald.copy(alpha = 0.9f), fontSize = 11.sp, fontWeight = FontWeight.SemiBold)
        }
        Row(
            modifier = Modifier.fillMaxWidth().height(10.dp).clip(RoundedCornerShape(50)),
        ) {
            Box(modifier = Modifier.weight(h.coerceAtLeast(1).toFloat()).fillMaxHeight().background(WcColors.emerald))
            Box(modifier = Modifier.weight(d.coerceAtLeast(1).toFloat()).fillMaxHeight().background(Color.White.copy(alpha = 0.55f)))
            Box(modifier = Modifier.weight(a.coerceAtLeast(1).toFloat()).fillMaxHeight().background(WcColors.sky))
        }
        Text("توقعات خوارزمية للاستئناس من مزود البيانات", color = Color.White.copy(alpha = 0.4f), fontSize = 10.sp)
    }
}

/** ترويسة قسم: أيقونة + عنوان + وصف (على خلفية داكنة). */
@Composable
fun WcSectionHeader(icon: androidx.compose.ui.graphics.vector.ImageVector, title: String, subtitle: String, tint: Color = WcColors.emerald) {
    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(12.dp), modifier = Modifier.fillMaxWidth()) {
        Box(
            modifier = Modifier.size(40.dp).clip(RoundedCornerShape(12.dp)).background(tint.copy(alpha = 0.16f)),
            contentAlignment = Alignment.Center,
        ) {
            Icon(icon, contentDescription = null, tint = tint, modifier = Modifier.size(22.dp))
        }
        Column {
            Text(title, color = WcColors.onDark, fontSize = 21.sp, fontWeight = FontWeight.Black)
            Text(subtitle, color = WcColors.onDarkDim, fontSize = 12.sp)
        }
    }
}

@Composable
fun WcLoading() {
    Box(modifier = Modifier.fillMaxWidth().padding(vertical = 32.dp), contentAlignment = Alignment.Center) {
        CircularProgressIndicator(color = Color.White, strokeWidth = 2.dp)
    }
}

@Composable
fun WcEmptyText(message: String) {
    Text(
        message,
        color = WcColors.onDarkDim,
        fontSize = 13.sp,
        modifier = Modifier.fillMaxWidth().padding(vertical = 28.dp),
        textAlign = androidx.compose.ui.text.style.TextAlign.Center,
    )
}

File: sabq app ios/sabq/Components/WorldCupComponents.swift
import SwiftUI

// MARK: - World Cup — مكونات مشتركة

/// صورة بعيدة (شعار منتخب / صورة لاعب) مع بديل أثناء التحميل.
/// تستخدم `CachedAsyncImage` (كاش ذاكرة NSCache + قرص 256MB + جلسة تحميل
/// موازية + تصغير عند فك الترميز) بدل `AsyncImage` العادي — فالشعارات وصور
/// اللاعبين صغيرة، ومع الكاش تظهر فورًا عند إعادة الظهور بدل التحميل المتأخّر.
struct WCRemoteImage: View {
    let url: String
    var contentMode: ContentMode = .fit
    /// الصور هنا صغيرة (شعار ≤50pt، صورة لاعب 46pt) فيكفي سقف 300px.
    var maxPixelSize: CGFloat = 300

    var body: some View {
        CachedAsyncImage(
            url: URL(string: url),
            contentMode: contentMode,
            maxPixelSize: maxPixelSize
        ) {
            Color.clear
        }
    }
}

/// شعار منتخب داخل دائرة بيضاء (الشعارات شفافة فتحتاج خلفية).
struct WCTeamLogo: View {
    let team: WCTeam
    var size: CGFloat = 40
    var ring: Color = Color.white.opacity(0.15)

    var body: some View {
        WCRemoteImage(url: team.logo)
            .padding(size * 0.14)
            .frame(width: size, height: size)
            .background(Circle().fill(.white))
            .overlay(Circle().stroke(ring, lineWidth: 2))
    }
}

/// شارة حالة المباراة: مباشر (نبض أحمر) / انتهت / وقت الانطلاق.
struct WCStatusPill: View {
    let fixture: WCFixture
    var onDark: Bool = false

    var body: some View {
        if fixture.status.live {
            HStack(spacing: 4) {
                Circle().fill(.white).frame(width: 5, height: 5)
                Text(elapsedText)
            }
            .font(SabqFonts.app(size: 11, weight: .medium))
            .foregroundStyle(.white)
            .padding(.horizontal, 8).padding(.vertical, 3)
            .background(Capsule().fill(WCTheme.liveRed))
        } else if fixture.status.finished {
            Text(fixture.status.label)
                .font(SabqFonts.app(size: 11, weight: .regular))
                .foregroundStyle(WCTheme.liveRed)
                .padding(.horizontal, 8).padding(.vertical, 3)
                .background(Capsule().fill(WCTheme.liveRed.opacity(onDark ? 0.20 : 0.12)))
        } else {
            Text(WCFormat.time(fixture))
                .font(SabqFonts.app(size: 11, weight: .medium))
                .foregroundStyle(WCTheme.emeraldDeep)
                .padding(.horizontal, 8).padding(.vertical, 3)
                .background(Capsule().fill(WCTheme.emerald.opacity(0.16)))
        }
    }

    private var elapsedText: String {
        let s = fixture.status
        // عدّاد المباراة يجري فعلًا في أشواط اللعب فقط (لا الاستراحة/الترجيح/التوقف)
        let running = ["1H", "2H", "ET", "LIVE"].contains(s.code) && s.elapsed != nil
        if running, let e = s.elapsed {
            // الوقت بدل الضائع: 90+8' بدل 90' المجمدة في أكثر دقائق المباراة توترًا
            let minute = (s.extra ?? 0) > 0 ? "\(e)+\(s.extra!)'" : "\(e)'"
            // اسم الشوط يسبق الدقيقة: «الشوط الأول · 23'»
            return s.label.isEmpty ? minute : "\(s.label) · \(minute)"
        }
        // الاستراحات/التوقف/الترجيح: نص الحالة («استراحة الشوطين») لا دقيقة مجمّدة
        return s.label.isEmpty ? "مباشر" : s.label
    }
}

/// عدّ تنازلي حي يتحرّك كل ثانية (TimelineView).
struct WCCountdownChips: View {
    let timestamp: Int

    var body: some View {
        TimelineView(.periodic(from: .now, by: 1)) { _ in
            let total = max(0, Double(timestamp) - Date().timeIntervalSince1970)
            if total <= 0 {
                // الموعد حان والمزود لم يرفع إشارة «حية» بعد — لا أصفار مجمدة
                HStack(spacing: 6) {
                    Circle()
                        .fill(WCTheme.emerald)
                        .frame(width: 8, height: 8)
                    Text("حان موعد الانطلاق — التغطية الحية تبدأ خلال لحظات")
                        .font(SabqFonts.app(size: 12, weight: .medium))
                        .foregroundStyle(WCTheme.onDark)
                }
            } else {
                let days = Int(total) / 86_400
                let hours = (Int(total) % 86_400) / 3_600
                let minutes = (Int(total) % 3_600) / 60
                let seconds = Int(total) % 60
                HStack(spacing: 8) {
                    chip(days, "يوم")
                    chip(hours, "ساعة")
                    chip(minutes, "دقيقة")
                    chip(seconds, "ثانية")
                }
                .environment(\.layoutDirection, .leftToRight)
            }
        }
    }

    private func chip(_ value: Int, _ label: String) -> some View {
        VStack(spacing: 1) {
            Text("\(value)")
                .font(SabqFonts.app(size: 18, weight: .semibold))
                .foregroundStyle(WCTheme.onDark)
            Text(label)
                .font(SabqFonts.app(size: 10))
                .foregroundStyle(WCTheme.emeraldDeep)
        }
        .frame(minWidth: 52)
        .padding(.vertical, 6)
        .background(RoundedRectangle(cornerRadius: 12, style: .continuous).fill(WCTheme.chipFill))
    }
}

/// شريط احتمالات الفوز الثلاثي.
struct WCProbabilityBar: View {
    let fixture: WCFixture
    let prediction: WCPrediction

    var body: some View {
        let total = max(1, prediction.home + prediction.draw + prediction.away)
        let h = Int(Double(prediction.home) / Double(total) * 100)
        let d = Int(Double(prediction.draw) / Double(total) * 100)
        let a = Int(Double(prediction.away) / Double(total) * 100)
        VStack(spacing: 6) {
            HStack {
                Text("فوز \(fixture.home.name) \(h)%")
                Spacer()
                Text("تعادل \(d)%").foregroundStyle(WCTheme.onDarkDim)
                Spacer()
                Text("فوز \(fixture.away.name) \(a)%")
            }
            .font(SabqFonts.app(size: 11, weight: .regular))
            .foregroundStyle(WCTheme.emeraldDeep)

            GeometryReader { geo in
                HStack(spacing: 0) {
                    Rectangle().fill(WCTheme.royal).frame(width: geo.size.width * CGFloat(h) / 100)
                    Rectangle().fill(WCTheme.onDarkDim.opacity(0.5)).frame(width: geo.size.width * CGFloat(d) / 100)
                    Rectangle().fill(WCTheme.sky)
                }
            }
            .frame(height: 10)
            .clipShape(Capsule())

            Text("توقعات خوارزمية للاستئناس من مزود البيانات")
                .font(SabqFonts.app(size: 10))
                .foregroundStyle(WCTheme.onDarkDim)
        }
    }
}

/// شريط الاحتمالات لبطاقة الهيرو. التوقع يأتي جاهزًا من overview.predictions
/// للمباراة المميّزة؛ وعند غيابه (البطاقة الثانية المتزامنة) نتراجع إلى
/// /world-cup/forecast/:id — نفس مصدر مركز المباراة، مطابق لمكوّن الويب
/// HeroPrediction — حتى تُظهر كل البطاقات توقعها لا المميّزة وحدها.
struct WCHeroPrediction: View {
    let fixture: WCFixture
    let prediction: WCPrediction?
    @State private var fetched: WCForecast?

    private var resolved: WCPrediction? {
        if let p = prediction { return p }
        if let ft = fetched?.fulltime {
            return WCPrediction(home: ft.home, draw: ft.draw, away: ft.away, advice: nil)
        }
        return nil
    }

    var body: some View {
        Group {
            if let p = resolved {
                WCProbabilityBar(fixture: fixture, prediction: p)
            }
        }
        .task(id: fixture.id) {
            if prediction == nil, !fixture.status.finished {
                fetched = try? await APIClient.shared.fetchWorldCupForecast(fixtureId: fixture.id)
            }
        }
    }
}

/// ترويسة قسم: أيقونة + عنوان + وصف.
struct WCSectionHeader: View {
    let icon: String
    let title: String
    let subtitle: String
    var tint: Color = WCTheme.emeraldDeep

    var body: some View {
        HStack(spacing: 12) {
            Image(systemName: icon)
                .font(SabqFonts.app(size: 20, weight: .semibold))
                .foregroundStyle(tint)
                .frame(width: 40, height: 40)
                .background(RoundedRectangle(cornerRadius: 12, style: .continuous).fill(tint.opacity(0.12)))
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(SabqFonts.headline(size: 21))
                    .foregroundStyle(WCTheme.onDark)
                Text(subtitle)
                    .font(SabqFonts.app(size: 12))
                    .foregroundStyle(WCTheme.onDarkDim)
            }
            Spacer(minLength: 0)
        }
    }
}

/// حالة فارغة لطيفة داخل بطاقة.
struct WCEmptyState: View {
    let icon: String
    let title: String
    let subtitle: String

    var body: some View {
        VStack(spacing: 8) {
            Image(systemName: icon)
                .font(SabqFonts.app(size: 30))
                .foregroundStyle(WCTheme.emeraldDeep)
            Text(title)
                .font(SabqFonts.subhead(size: 15))
                .foregroundStyle(WCTheme.onDark)
            Text(subtitle)
                .font(SabqFonts.app(size: 12))
                .foregroundStyle(WCTheme.onDarkDim)
                .multilineTextAlignment(.center)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 28)
    }
}

/// مؤشر تحميل بلون العلامة. الأخضر العميق لا الأبيض: يُعرض غالبًا على
/// sectionBackground شبه الأبيض في النمط الفاتح، وكان الأبيض غير مرئي عمليًا.
struct WCLoading: View {
    var body: some View {
        HStack { Spacer(); ProgressView().tint(WCTheme.emeraldDeep); Spacer() }
            .padding(.vertical, 32)
    }
}
